client/me/purchases/manage-purchase/index.jsx

```diff
@@ -223,11 +225,39 @@ class ManagePurchase extends Component {
 		);
 	}
 
+	renderUpgradeButton() {
+		const { purchase, translate } = this.props;
+
+		const isUpgradeablePlan =
+			isPlan( purchase ) &&
```

Comment:
Noting that all this logic that builds `isUpgradeablePlan` is used a couple of times in the component and can be abstracted to a separate method.